pyjsonpatch/generate.py

```diff
@@ -22,18 +22,22 @@ def generate_patch(source: Any, target: Any) -> list[Operation]:
     patch: list[Operation] = []
 
     def _generate(source_: Any, target_: Any, path: str):
-        if source_ == target_:
+        if source is target_ or source_ == target_:
```

Comment:
Could just pull the `_generate` out of this function so the params don't need trailing underscores. Would also be technically faster since every call to `generate` is going to create a new `_generate`

```suggestion
        if source_ is target_ or source_ == target_:
```